model-server/src/test/kotlin/org/modelix/model/server/BulkSyncPerformanceTest.kt

```diff
@@ -0,0 +1,108 @@
+import io.ktor.server.resources.Resources
+import io.ktor.server.routing.IgnoreTrailingSlash
+import io.ktor.server.testing.ApplicationTestBuilder
+import io.ktor.server.testing.testApplication
+import io.ktor.server.websocket.WebSockets
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.coroutineScope
+import org.modelix.authorization.installAuthentication
+import org.modelix.model.api.IChildLink
+import org.modelix.model.api.IConceptReference
+import org.modelix.model.api.INode
+import org.modelix.model.api.addNewChild
+import org.modelix.model.api.getRootNode
+import org.modelix.model.client2.ModelClientV2
+import org.modelix.model.client2.runWriteOnBranch
+import org.modelix.model.lazy.RepositoryId
+import org.modelix.model.server.handlers.KeyValueLikeModelServer
+import org.modelix.model.server.handlers.ModelReplicationServer
+import org.modelix.model.server.handlers.RepositoriesManager
+import org.modelix.model.server.store.InMemoryStoreClient
+import org.modelix.model.server.store.LocalModelClient
+import kotlin.random.Random
+import kotlin.test.Test
+import kotlin.test.assertTrue
+
+class BulkSyncPerformanceTest {
+    private lateinit var repositoriesManager: RepositoriesManager
+    private fun runTest(block: suspend ApplicationTestBuilder.(scope: CoroutineScope) -> Unit) = testApplication {
+        application {
+            installAuthentication(unitTestMode = true)
+            install(ContentNegotiation) {
+                json()
+            }
+            install(WebSockets)
+            install(Resources)
+            install(IgnoreTrailingSlash)
+            repositoriesManager = RepositoriesManager(LocalModelClient(InMemoryStoreClient()))
+            ModelReplicationServer(repositoriesManager).init(this)
+            KeyValueLikeModelServer(repositoriesManager).init(this)
+        }
+
+        coroutineScope {
+            block(this)
+        }
+    }
+
+    /**
+     * Tests the performance of the `GET /v2/repositories/{repository}/branches/{branch}` endpoint.
+     * Many small request to an IgniteStoreClient lead to a poor performance. This test ensure that bulk requests are
+     * used for loading a model.
+     */
+    @Test
+    fun `efficient bulk sync`() = runTest {
+        val rand = Random(1056343)
+        val url = "http://localhost/v2"
+        val modelClient = ModelClientV2.builder().url(url).client(client).build().also { it.init() }
+        val repositoryId = RepositoryId("repo1")
+        modelClient.initRepository(repositoryId)
+
+        modelClient.runWriteOnBranch(repositoryId.getBranchReference()) { branch ->
+            val rootNode = branch.getRootNode()
+            repeat(10_000) {
+                val randomNode = rootNode.getRandomNode(rand)
+                randomNode.addNewChild(IChildLink.fromName("roleA"), -1, null as IConceptReference?)
+            }
+        }
+
+        val modelClient2 = ModelClientV2.builder().url(url).client(client).build().also { it.init() }
```

Comment:
```suggestion
        val requestingModelClient = ModelClientV2.builder().url(url).client(client).build().also { it.init() }
```

Or something else that speaks a bit more than "2".

Reply:
Done